### PR TITLE
Implement shortnames feature

### PR DESCRIPTION
**Pull request by ericcurtin**
Add support for resolving model shortnames. Parse shortnames from /etc/ramalama/registries.conf.d/000-shortnames.conf . Modify New() function to use resolved shortnames. Improve user experience by allowing use of familiar shorthand names for models. This change allows users to use shortnames instead of full model names, similar to podman functionality. The feature enhances usability while maintaining compatibility with existing model naming conventions.

### CODE REVIEW

File: ramalama/cli.py [hunk 1]
```diff
@@ -467,7 +468,23 @@ def run_container(args):
     exec_cmd(conman_args)
 
 
+def load_shortnames():
+    config = configparser.ConfigParser()
+    config_file = "/etc/ramalama/registries.conf.d/000-shortnames.conf"
```
Comment (ericcurtin) [inline ramalama/cli.py:473]: I should probably iterate through all the files in "/etc/ramalama/registries.conf.d/"

**Comment by ericcurtin** [comment]
@rhatdan it's not clear to me how podman/bats handles tests that depend on certain files like "/etc/containers/registries.conf.d/000-shortnames.conf"

I can see tests in podman/bats testing the shortnames features, but they seem to test failure cases, rather testing successful cases.

**Comment by ericcurtin** [comment]
Linked issue: https://github.com/containers/ramalama/issues/38

**Comment by ericcurtin** [comment]
I still have to do directory precedence style code here. For example on macOS rootless installs are more common, so probably don't want to edit /etc at all:

etc/ramalama/registries.conf.d/000-shortnames.conf

File: etc/ramalama/registries.conf.d/000-shortnames.conf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+[aliases]
+  "granite" = "huggingface://instructlab/granite-7b-lab-GGUF/granite-7b-lab-Q4_K_M.gguf"
```
Comment (rhatdan) [inline etc/ramalama/registries.conf.d/000-shortnames.conf:2]: Should add multiple
granite7b (and or granite-7b) ( I would love it if we stored these in cri-o by default, will talk about this next week)
granite-lab

Comment (rhatdan) [reply etc/ramalama/registries.conf.d/000-shortnames.conf:2]: This file should also be in /usr/share.  Leave /etc for sysadmins. /usr/share for distributions.

File: ramalama/cli.py [hunk 3]
```diff
@@ -439,6 +451,7 @@ def run_container(args):
         f"-v{home}:{home}",
         f"-v{sys.argv[0]}:/usr/bin/ramalama:ro",
         f"-v{wd}:/usr/share/ramalama/ramalama:ro",
+        f"-v{ed}:/etc/ramalama:ro",
```
Comment (rhatdan) [inline ramalama/cli.py:454]: We got to think about this one.

**Comment by rhatdan** [comment]
We want to read these files in /usr/share/ramalama/aliases.conf, /etc/ramalama/aliases.conf $HOME/.config/ramalama/aliases.conf

For the in the container use case we should prepare a file out of this content with the last ALIASES winning and then mount it into the container.

cat  /usr/share/ramalama/aliases.conf /etc/ramalama/aliases.conf $HOME/.config/ramalama/aliases.conf > /tmp/aliases.conf and then process it, removing duplicates.

**Comment by ericcurtin** [comment]
Makes sense to me, although instead of this one:

`$HOME/.config/ramalama/aliases.conf`

we could do:

`$HOME/.local/share/ramalama/aliases.conf`

`$HOME/.local/share/ramalama` is a directory we already use and it kinda fits with `/usr/share/ramalama/aliases.conf `

But I'm easy about directory structures, happy to go completely as you described also @rhatdan ...

File: etc/ramalama/registries.conf.d/000-shortnames.conf (hunk 2 above)
Comment (ericcurtin) [reply etc/ramalama/registries.conf.d/000-shortnames.conf:2]: What ollama does... Which is kinda interesting is this format (like container images):

granite-code:8b
granite-code:3b
granite-code:latest (alias for 3b tag or whatever tag one selects)

Comment (rhatdan) [reply etc/ramalama/registries.conf.d/000-shortnames.conf:2]: Yup that would make sense.

**Comment by rhatdan** [comment]
Well $HOME/.local/share is for app content, whereas $HOME/.config/ is for user custom configuration.

As far as the mount within the container,  I think we will need to process all of the aliases on the users host system into one aliases.conf file which we mount into the container.  So assemble it from /usr, /etc/ and $HOME into something in /tmp and then mount into container.  I don't think we want to mount all of the mounts from the host into the container. This way the container itself could have some aliases.conf files which are not on the users host system.

File: etc/ramalama/registries.conf.d/000-shortnames.conf (hunk 2 above)
Comment (ericcurtin) [reply etc/ramalama/registries.conf.d/000-shortnames.conf]: At the moment I have:

"granite-lab-7b" and  "merlinite-lab-7b"

the ':' characters don't parse well in the .conf file with the python parser, but can try again to make ':' work if we really want the ':' character to be present.

**Comment by ericcurtin** [comment]
> Well $HOME/.local/share is for app content, whereas $HOME/.config/ is for user custom configuration.
> 
> As far as the mount within the container, I think we will need to process all of the aliases on the users host system into one aliases.conf file which we mount into the container. So assemble it from /usr, /etc/ and $HOME into something in /tmp and then mount into container. I don't think we want to mount all of the mounts from the host into the container. This way the container itself could have some aliases.conf files which are not on the users host system.

I moved the resolving of short-names to long-names to occur outside of the container, so now no temporary files required, seemed cleaner... Less difference between container and no-container case and no temporary files to clean up. And now no bind mounts also.

File: ramalama/ollama.py [hunk 4]
```diff
@@ -41,7 +41,7 @@ def init_pull(repos, manifests, accept, registry_head, model_name, model_tag, mo
             manifest_data = json.load(f)
     except subprocess.CalledProcessError as e:
         if e.returncode == 22:
-            raise KeyError((f"{model}:{model_tag} not found"))
+            raise KeyError((f"{model} not found"))
```
Comment (ericcurtin) [inline ramalama/ollama.py:44]: {model} already has {model_tag}  so it made the output look weird, duplicate tag at the end

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**